code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -322,6 +322,14 @@
 /mob/living/critter/flock/drone/Life(datum/controller/process/mobs/parent)
 	if (..(parent))
 		return 1
+	if (src.floorrunning && src.resources)
```

Comment:
```suggestion
	if (src.floorrunning && (src.resources > 0))
```
negative numbers aren't falsey, and I forsee an issue where someone manages to get resources to be negative and do infinite floorrunning